src/value-parser.js

```diff
@@ -335,7 +335,8 @@ function valueParse(parser, metaData, options, callback) {
                     valueMetaData.collation = collation;
                     if (baseType === 0xA5 || baseType === 0xAD || baseType === 0xA7 || baseType === 0xAF || baseType === 0xE7 || baseType === 0xEF) {
                       return readDataLength(parser, valueMetaData.type, {}, null, (maxDataLength) => {
-                        valueMetaData.dataLength = maxDataLength;
+                        // skip the 2-byte max length sent for BIGVARCHRTYPE, BIGCHARTYPE, NVARCHARTYPE, NCHARTYPE, BIGVARBINTYPE and BIGBINARYTYPE types
+                        // and parse based on the length of actual data
```

Comment:
Thanks for clarifying that! 

Passing maxDataLength and using it as MAX doesn't always work. For instance, `select cast('abc' as sql_variant);` returns maxDataLength as 3, in this case `(metaData.dataLength === MAX)` so flow will enter into `readMaxChars` to read PLP stream. `2-byte max length` mentioned in the TDS is flaky, changes based on SQL.